### PR TITLE
SPARK-15641 HistoryServer to not show invalid date for incomplete application

### PR DESCRIPTION
**Pull request by catapan**
## What changes were proposed in this pull request?

For incomplete applications in HistoryServer, the complete column will show "-" instead of incorrect date.
## How was this patch tested?

manually tested.

### CODE REVIEW

File: core/src/main/resources/org/apache/spark/ui/static/historypage.js [hunk 1]
```diff
@@ -54,7 +54,8 @@ function makeIdNumeric(id) {
 }
 
 function formatDate(date) {
-  return date.split(".")[0].replace("T", " ");
+  if (date <= 0) return "-";
```
Comment (srowen) [inline core/src/main/resources/org/apache/spark/ui/static/historypage.js]: I suppose I don't have a preference on "" vs "-". LGTM

**Comment by srowen** [comment]
Jenkins test this please

File: core/src/main/resources/org/apache/spark/ui/static/historypage.js (hunk 1 above)
Comment (tejasapatil) [inline core/src/main/resources/org/apache/spark/ui/static/historypage.js]: How about using "n/a" which would specifically indicate that its invalid to expect the end time ?

**Comment by tejasapatil** [comment]
nit: the title of the PR could be made better:

`[SPARK-15641] HistoryServer to not show invalid date for incomplete application`

**Comment by catapan** [comment]
revised as suggested. Thanks for the suggestions.

File: core/src/main/resources/org/apache/spark/ui/static/historypage.js (hunk 1 above)
Comment (srowen) [inline core/src/main/resources/org/apache/spark/ui/static/historypage.js]: _shrug_ I think blank or - is simpler and more general. I guess n/a takes a moment to parse as not applicable.

Comment (catapan) [inline core/src/main/resources/org/apache/spark/ui/static/historypage.js]: How about I spell it out as "Not available"? blank or "-" seems not very informative to the users.

Comment (srowen) [inline core/src/main/resources/org/apache/spark/ui/static/historypage.js]: I personally think that just adds more noise to the screen. Sometimes the column value is a date, sometimes a text string then. Using a blank is the most natural "no value" to me.

Comment (catapan) [inline core/src/main/resources/org/apache/spark/ui/static/historypage.js]: I see. But to me, blank indicates that something is wrong. "-" is a little better. So I changed it back to "-".

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**